tfhe/src/integer/server_key/radix_parallel/tests_cases_unsigned.rs

```diff
@@ -0,0 +1,2825 @@
+        assert!(ctxt_condition.holds_boolean_value());
+
+        let ct_res = executor.execute((&ctxt_condition, &ctxt_0, &ctxt_1));
+        assert!(ct_res.block_carries_are_empty());
+
+        let dec_res: u64 = cks.decrypt(&ct_res);
+        assert_eq!(
+            dec_res,
+            if clear_condition == 1 {
+                clear_0
+            } else {
+                clear_1
+            }
+        );
+
+        let ct_res2 = executor.execute((&ctxt_condition, &ctxt_0, &ctxt_1));
+        assert_eq!(ct_res, ct_res2, "Operation is not deterministic");
+
+        let clear_2 = rng.gen::<u64>() % modulus;
+        let clear_3 = rng.gen::<u64>() % modulus;
+
+        let ctxt_2 = cks.encrypt(clear_2);
+        let ctxt_3 = cks.encrypt(clear_3);
+
+        // Add to have non empty carries
+        sks.unchecked_add_assign(&mut ctxt_0, &ctxt_2);
+        sks.unchecked_add_assign(&mut ctxt_1, &ctxt_3);
+        assert!(!ctxt_0.block_carries_are_empty());
+        assert!(!ctxt_1.block_carries_are_empty());
+
+        let ct_res = executor.execute((&ctxt_condition, &ctxt_0, &ctxt_1));
+        assert!(ct_res.block_carries_are_empty());
+
+        let dec_res: u64 = cks.decrypt(&ct_res);
+        assert_eq!(
+            dec_res,
+            if clear_condition == 1 {
+                (clear_0 + clear_2) % modulus
+            } else {
+                (clear_1 + clear_3) % modulus
+            }
+        );
+    }
+}
+
+//=============================================================================
+// Default Scalar Tests
+//=============================================================================
+
+pub(crate) fn default_scalar_add_test<P, T>(param: P, mut executor: T)
+where
+    P: Into<PBSParameters>,
+    T: for<'a> FunctionExecutor<(&'a RadixCiphertext, u64), RadixCiphertext>,
+{
+    let (cks, mut sks) = KEY_CACHE.get_from_params(param);
+    let cks = RadixClientKey::from((cks, NB_CTXT));
+
+    sks.set_deterministic_pbs_execution(true);
+
+    // message_modulus^vec_length
+    let modulus = cks.parameters().message_modulus().0.pow(NB_CTXT as u32) as u64;
+
+    let mut rng = rand::thread_rng();
+
+    executor.setup(&cks, sks.clone());
+
+    let mut clear;
+
+    for _ in 0..NB_TEST_SMALLER {
+        let clear_0 = rng.gen::<u64>() % modulus;
+        let clear_1 = rng.gen::<u64>() % modulus;
+
+        // encryption of an integer
+        let ctxt_0 = cks.encrypt(clear_0);
+
+        let mut ct_res = executor.execute((&ctxt_0, clear_1));
+        assert!(ct_res.block_carries_are_empty());
+
+        clear = (clear_0 + clear_1) % modulus;
+
+        // Add multiple times to raise the degree
+        for _ in 0..NB_TEST_SMALLER {
+            let tmp = executor.execute((&ct_res, clear_1));
+            ct_res = executor.execute((&ct_res, clear_1));
+            assert!(ct_res.block_carries_are_empty());
+            assert_eq!(ct_res, tmp);
+            clear = (clear + clear_1) % modulus;
```

Comment:
maybe worth using wrapping ops, but as the values are small we should not have an issue for now